### ISSUE TITLE
dockerng.running: if I remove an enviroment variable the container does not restart

### ISSUE THREAD

**Issue by mirkofranzil** [saltstack/salt#39191]
### Description of Issue/Question
If I remove an environment variable dockerng.running doesn't restart container

### Steps to Reproduce Issue
I'm starting with this:
```
foo:
  dockerng.running:
    - image: bar/baz:latest
    - environment:
      - VAR1: value1
      - VAR2: value2
```
If I comment or I remove an environment variable:
```
foo:
  dockerng.running:
    - image: bar/baz:latest
    - environment:
      # - VAR1: value1
      - VAR2: value2
```
dockerng.running doesn't restart the container .
It restarts only if I change the variable data like this:
```
foo:
  dockerng.running:
    - image: bar/baz:latest
    - environment:
      - VAR1: ""
      - VAR2: value2
```
There is no check on deleted environment vars but only if data changes in listed environment vars.

### Versions Report
```
Salt Version:
           Salt: 2016.3.5
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.0
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.4.0-62-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by gtmanfred** [comment]
Thanks for reporting this.

I have tagged our platform team to take a look when they get a chance, If you wanted to try fixing it, we would greatly appreciate a pull request.

Thanks!
Daniel

**Comment by terminalmage** [comment]
This was fixed in 2017.7.0. Closing.
